src/main/java/com/volasoftware/tinder/services/implementations/VerificationTokenServiceImpl.java

```diff
@@ -53,6 +54,26 @@ public Account verifyToken(String token) {
         return updatedVerificationToken.getAccount();
     }
 
+    @Override
+    public VerificationToken regenerateToken(Account account) {
+        Optional<VerificationToken> optionalVerificationToken = verificationTokenRepository
+            .findByAccountId(account.getId());
+
+        if (optionalVerificationToken.isEmpty()) {
+            return generateToken(account);
+        }
+
+        VerificationToken verificationToken = optionalVerificationToken.get();
+        String uuidToken = UUID.randomUUID().toString();
+        LocalDateTime currentTime = LocalDateTime.now();
+        verificationToken.setToken(uuidToken);
+        verificationToken.setLastModifiedDate(currentTime);
+        verificationToken.setExpiresAt(currentTime.plusDays(SecurityConstant.TOKEN_EXPIRATION_DAYS)
+        );
+
+        return verificationTokenRepository.save(verificationToken);
+    }
```

Comment:
What is happening with the old token?

Reply:
If the account is not verified and there is already token, the record will be updated in the database.